promql/test.go

```diff
@@ -771,6 +778,10 @@ func (t *test) execRangeEval(cmd *evalCmd, engine QueryEngine) error {
 	res := q.Exec(t.context)
 	if res.Err != nil {
 		if cmd.fail {
+			if cmd.expectedFailMessage != "" && cmd.expectedFailMessage != res.Err.Error() {
```

Comment:
could we put this failure check block (779-792) into a `checkFailure` or sth else.
As it's the same code as in `execInstantEval`

Reply:
Makes sense to me, I've fixed this in 4ed4787.